velox/functions/prestosql/StringFunctions.h

```diff
@@ -267,4 +267,88 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+template <typename T>
+struct LevenshteinDistanceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    std::vector<int32_t> leftCodePoints =
+        stringImpl::stringToCodePoint<false, int32_t>(left);
+    std::vector<int32_t> rightCodePoints =
+        stringImpl::stringToCodePoint<false, int32_t>(right);
+    doCall<int32_t>(result, leftCodePoints, rightCodePoints);
+  }
+
+  FOLLY_ALWAYS_INLINE void callAscii(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    std::vector<int8_t> leftCodePoints =
+        stringImpl::stringToCodePoint<true, int8_t>(left);
+    std::vector<int8_t> rightCodePoints =
+        stringImpl::stringToCodePoint<true, int8_t>(right);
+    doCall<int8_t>(result, leftCodePoints, rightCodePoints);
+  }
+
+  template <typename codePointT>
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<int64_t>& result,
+      std::vector<codePointT> leftCodePoints,
+      std::vector<codePointT> rightCodePoints) {
+    if (leftCodePoints.size() < rightCodePoints.size()) {
+      leftCodePoints.swap(rightCodePoints);
+    }
+    if (rightCodePoints.empty()) {
+      result = leftCodePoints.size();
+      return;
+    }
+
+    static const int32_t kMaxCombinedInputSize = 1'000'000;
+    int32_t combinedInputSize =
+        leftCodePoints.size() * (rightCodePoints.size() - 1);
+    VELOX_USER_CHECK_LE(
+        combinedInputSize,
+        kMaxCombinedInputSize,
+        "The combined inputs size {} exceeded max Levenshtein distance combined input size {}."
+        " The code points size of left is {}, code points size of right is {}",
+        combinedInputSize,
```

Comment:
No need to include combinedInputSize and kMaxCombinedInputSize in the error message. These are included automatically.